src/components/UpdateCapRover.svelte

```diff
@@ -0,0 +1,381 @@
+  import getGrid from "../utils/getGrid";
+  const { AddMachineModel, DeleteMachineModel, DiskModel } = window.configs?.grid3_client ?? {}; // prettier-ignore
+
+  const dispatch = createEventDispatcher<{ closed: boolean }>();
+
+  export let profile: IProfile;
+  export let capRover: any;
+
+  let workers: any[] = [];
+
+  let shouldBeUpdated: boolean = false;
+  let loading: boolean = false;
+  let workers_loading: boolean = true;
+  let message: string;
+  let success: boolean = false;
+  let failed: boolean = false;
+  let removing: string = null;
+  let domain: string = "";
+
+  let worker = new CapWorker();
+  let grid;
+
+  let diskField: IFormField;
+  let cpuField: IFormField;
+  let memoryField: IFormField;
+
+  const CAPROVER_FLIST = "https://hub.grid.tf/tf-official-apps/tf-caprover-main.flist";
+  
+  // prettier-ignore
+  const workerFields: IFormField[] = [ 
+    { label: "Name", symbol: "name", placeholder: "Worker instance name", type: "text", validator: validateName, invalid: false },
+  ];
+
+  const packages: IPackage[] = [
+    { name: "Minimum", cpu: 1, memory: 1024, diskSize: 50 },
+    { name: "Standard", cpu: 2, memory: 1024 * 2, diskSize: 100 },
+    { name: "Recommended", cpu: 4, memory: 1024 * 4, diskSize: 250 },
+  ];
+
+  $: workerData = false;
+  $: workerIp = "";
+  $: disabled = loading || isInvalid([...workerFields]) || !worker.valid; // prettier-ignore
+  $: logs = $currentDeployment;
+
+  onMount(async () => {
+    worker.publicKey = capRover.details.env.PUBLIC_KEY;
+    domain = capRover.details.env.CAPROVER_ROOT_DOMAIN;
+
+    grid = await getGrid(profile, (grid) => grid, false);
+    grid.projectName = "caprover";
+    grid._connect();
+
+    if (capRover) workers = 
+      (await grid.machines.getObj(capRover["name"])).filter((machine) => machine.env["SWM_NODE_MODE"] == "worker");
+    
+    workers_loading = false;
+  });
+
+  async function onAddWorker() {
+    loading = true;
+    currentDeployment.deploy("Add Worker", worker.name);
+    
+    /* Docker disk */
+    const disk = new DiskModel();
+    disk.name = "data0";
+    disk.size = worker.diskSize;
+    disk.mountpoint = "/var/lib/docker";
+
+    const workerModel = new AddMachineModel();
+    workerModel.deployment_name = capRover["name"];
+    workerModel.cpu = worker.cpu;
+    workerModel.memory = worker.memory;
+    workerModel.disks = [disk];
+    workerModel.node_id = worker.nodeId;
+    workerModel.public_ip = true;
+    workerModel.name = `CRW${worker.name}`;
+    workerModel.planetary = false;
+    workerModel.flist = CAPROVER_FLIST;
+    workerModel.qsfs_disks = [];
+    workerModel.rootfs_size = rootFs(worker.cpu, worker.memory);
+    workerModel.entrypoint = "/sbin/zinit init";
+    workerModel.env = {
+      SWM_NODE_MODE: "worker",
+      LEADER_PUBLIC_IP: capRover["publicIp"].split("/")[0],
+      CAPTAIN_IMAGE_VERSION: "latest",
+      PUBLIC_KEY: worker.publicKey,
+      CAPTAIN_IS_DEBUG: "true",
```

Comment:
no need for these environment variables (CAPTAIN_IS_DEBUG, CAPTAIN_IMAGE_VERSION, and LEADER_PUBLIC_IP)